website/src/models/group.js

```diff
@@ -352,6 +354,77 @@ GroupSchema.methods.toJSON = function() {
   return doc;
 };
 
+GroupSchema.statics.removeUserFromChallenges = function(user, group, keep, cb) {
```

Comment:
Thanks for the example. I was definitely thinking of separating the functions out like that. I will make the changes both of you recommended.